docker-compose.yml

```diff
@@ -0,0 +1,47 @@
+---
+version: '2'
+
+services:
+  zookeeper:
+    image: confluentinc/cp-zookeeper:7.1.0
+    hostname: zookeeper
+    container_name: zookeeper
+    ports:
+      - "2181:2181"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
+
+  broker:
+    image: confluentinc/cp-kafka:7.1.0
+    hostname: broker
+    container_name: broker
+    depends_on:
+      - zookeeper
+    ports:
+      - "29092:29092"
+    environment:
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:9092,PLAINTEXT_HOST://localhost:29092
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
+      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
+
+  hazelcast:
+    image: hazelcast/hazelcast:5.2-BETA-1
+    hostname: hazelcast
+    container_name: hazelcast
+    ports:
+    - "5701:5701"
```

Comment:
This publishes the port to everyone, not only localhost. We should stop doing that.

cc @kwart

Reply:
Completely agree.
When the port exposure is necessary, we should use advanced networking and only pick port numbers of the required protocols (REST or client, for instance).

Reply:
We don't need to expose the ports in this example, so I'll remove this line.

Is it still OK to expose MC?

Reply:
Safer would be to expose it only for the loopback interface. I.e. instead of the `"8080:8080"` use the  `"127.0.0.1:8080:8080"`.